mat/svd.go

```diff
@@ -147,6 +147,16 @@ func (svd *SVD) Kind() SVDKind {
 	return svd.kind
 }
 
+// Rank returns the rank of A based on the count of singular values greater than cond.
+func (svd *SVD) Rank(cond float64) int {
```

Comment:
It should be possible to return the rank as part of the SVD. Depending on the underlying algorithm, this is likely to be `m*n*epsilon`, where `m` and `n` are the dimensions of A, or `s(1)*epsilon`. See http://www.netlib.no/netlib/lapack/lug/node96.html.

Naturally, if input data was truncated/quantized with a smaller precision, the value based on the machine epsilon will not be valid. In this context, this `Rank` function is useful, but I think that `cond` should be the quantization precision, and the function should take the dimensions of the problem into account.

Reply:
2. Seems reasonable.